### ISSUE TITLE
Allow any text node to contain expressions

### ISSUE THREAD

**Issue by danielfernandez** [thymeleaf/thymeleaf#307]
Thymeleaf would be a bit easier to understand and use for people coming from textual template engines like JSP, FreeMarker or Velocity if we were able to allow Thymeleaf Standard Expressions anywhere in Text nodes. Something like:

``` html
<p>
   Hello, ${user.name}
</p>
```

All kinds of simple expressions (`${...}`, `#{...}` and `@{...}`) and conversion expressions (`${{...}}`, `#{{...}}` and `@{{...}}`) could be embedded just like seen above, and the `[[...]]` syntax currently used for `th:inline="text"` could be used for complex (e.g. conditional) expressions:

``` html
<p>
   Hello, [[${user.isAdmin} ? #{salute.sir} : ${user.name}]]
</p>
```

Of course, this would only be feasible if we are able to find and parse expressions in text nodes at negligible computation cost during parsing. And using this feature **would be not recommended from the point of view of style**, as it would go against the idea of _Natural Templates_.

Note this would render `th:inline="text"` unnecessary.

**Comment by danielfernandez** [comment]
This has been implemented in Thymeleaf 3.0 by means of the new support for Inlined Output Expressions. See #394 and #396

**Comment by danielfernandez** [comment]
Already in `3.0.0-SNAPSHOT`
